RELEASES.md

```diff
@@ -1,3 +1,13 @@
+Version 1.72.1 (2023-09-14)
```

Comment:
Mind fixing the date here to the actual release, 9-19?

r=me with that done and commits squashed.

Reply:
OK. I updated the date.

What is r=me? (Is it a rustbot command?)